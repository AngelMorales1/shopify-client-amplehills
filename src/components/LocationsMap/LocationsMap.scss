@import 'settings';

.LocationsMap {
  height: calc(100vh - #{$header-height-desktop});

  &--with-alert {
    height: calc(100vh - #{$header-height-desktop} - #{$alert-height-desktop});
  }

  &__map-key-item {
    height: $button-small-height;
    border-radius: radius('l');

    span {
      margin-top: spacing(1);
    }
  }

  &__seasonal-button--selected {
    box-shadow: $drop-shadow;
    transform: translate3d(0, spacing(-0.25), 0);
  }

  &__circle-icon {
    width: spacing(6);
    height: spacing(6);
    font-size: 0.4rem;
  }
}
