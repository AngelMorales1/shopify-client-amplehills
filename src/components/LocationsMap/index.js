import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import LocationsMapFilters from 'constants/LocationsMapFilters';
import LocationsMapKey from 'constants/LocationsMapKey';

import { Image, Button } from 'components/base';
import MapboxMap from 'components/MapboxMap';
import styles from './LocationsMap.scss';

const LocationsMap = props => {
  const { filteredOutLocations, locationFilters, actions } = props;

  return (
    <div className={cx(styles['LocationsMap'], 'relative')}>
      <MapboxMap
        className="z-0"
        featureCollection={props.locationGeoJSON}
        defaultIcon="year-round-icon"
        styleUrl="mapbox://styles/joshiefishbein/cjjyuj8fq0hrj2ro2j8066e4q"
        collections={[
          {
            name: 'Selected',
            filter: {
              ids: []
            },
            icon: 'selected-location-icon'
          },
          {
            name: 'SeasonalLocations',
            filter: {
              ids: props.locations
                .filter(location => location.seasonal)
                .map(location => location.id)
            },
            icon: 'seasonal-icon'
          },
          {
            name: 'HiddenByFilter',
            filter: {
              ids: filteredOutLocations
            },
            visible: false
          }
        ]}
        iconSize={1.4}
        cluster={true}
        clusterMaxZoom={14}
        clusterRadius={50}
        textSize={16}
        textColor="#ffffff"
        mapPadding={150}
      />
      <div className="absolute t0 l0 flex p3">
        <Button
          className="mr2 flex items-center drop-shadow"
          color={locationFilters.length ? 'white-denim' : 'madison-blue'}
          variant="primary-small"
          label="All"
          onClick={actions.clearLocationFilters}
        />
        {LocationsMapFilters.STATE_FILTERS.map(filter => {
          const filterIsActive = locationFilters.some(
            activeFilter =>
              activeFilter.key === filter.key &&
              activeFilter.value === filter.value
          );

          return (
            <Button
              className="mr2 flex items-center drop-shadow"
              color={filterIsActive ? 'madison-blue' : 'white-denim'}
              variant="primary-small"
              key={filter.value}
              label={filter.label}
              onClick={
                filterIsActive
                  ? () =>
                      actions.removeLocationFilter({
                        key: filter.key,
                        value: filter.value
                      })
                  : () =>
                      actions.addLocationFilter({
                        key: filter.key,
                        value: filter.value
                      })
              }
            />
          );
        })}
      </div>
      <div className="absolute b0 l0 flex p3">
        {LocationsMapKey.SEASONAL_FILTERS.map(filter => (
          <div
            className={cx(
              styles['LocationsMap__map-key-item'],
              'mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold'
            )}
            key={filter.value}
          >
            <Image src={filter.icon} className="mr2" />
            <span>{filter.label}</span>
          </div>
        ))}
      </div>
    </div>
  );
};

LocationsMap.propTypes = {
  locationGeoJSON: PropTypes.shape({
    type: PropTypes.string,
    features: PropTypes.arrayOf(PropTypes.object)
  })
};

export default LocationsMap;
