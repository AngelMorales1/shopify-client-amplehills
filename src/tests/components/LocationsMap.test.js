import React from 'react';
import { shallow } from 'enzyme';

import LocationsMap from 'components/LocationsMap';

it('renders without data', () => {
  const component = shallow(<LocationsMap />);

  expect(component).toMatchSnapshot();
});
