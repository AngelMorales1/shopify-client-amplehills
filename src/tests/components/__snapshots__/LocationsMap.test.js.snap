// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationsMap
    actions={
        Object {
            "addLocationFilter": [Function],
            "clearLocationFilters": [Function],
            "clearLocationSelection": [Function],
            "removeLocationFilter": [Function],
            "selectLocation": [Function],
          }
    }
    locationFilters={Array []}
    locationGeoJSON={Object {}}
    locations={Array []}
    states={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MapboxMap
          className="z-0"
          cluster={true}
          clusterMaxZoom={14}
          clusterRadius={50}
          collections={
                    Array [
                              Object {
                                "filter": Object {
                                  "ids": Array [],
                                },
                                "icon": "selected-location-icon",
                                "name": "Selected",
                              },
                              Object {
                                "filter": Object {
                                  "ids": Array [],
                                },
                                "icon": "seasonal-icon",
                                "name": "SeasonalLocations",
                              },
                              Object {
                                "filter": Object {
                                  "ids": undefined,
                                },
                                "name": "HiddenByFilter",
                                "visible": false,
                              },
                            ]
          }
          defaultIcon="year-round-icon"
          featureCollection={Object {}}
          featureIdZoomTo={undefined}
          hoverFade={false}
          iconSize={1.4}
          mapPadding={150}
          maxZoom={18}
          onClickFeature={[Function]}
          onLoad={[Function]}
          styleUrl="mapbox://styles/joshiefishbein/cjjyuj8fq0hrj2ro2j8066e4q"
          textColor="#ffffff"
          textSize={16}
/>,
        <div
          className="absolute t0 l0 flex p3"
>
          <Button
                    className="mr2 flex flex-wrap items-center drop-shadow"
                    color="madison-blue"
                    disabled={false}
                    fullWidth={false}
                    hover="shadow"
                    label="All"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    shadow={false}
                    to=""
                    type="button"
                    variant="primary-small"
          />
</div>,
        <div
          className="absolute b0 l0 flex p3"
>
          <div
                    className="mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold"
          >
                    <Image
                              alt=""
                              bg={false}
                              className="mr2"
                              onImgLoad={[Function]}
                              src="/assets/images/year-round-icon-color.svg"
                              style={Object {}}
                    />
                    <span>
                              Open Year Round
                    </span>
          </div>
          <div
                    className="mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold"
          >
                    <Image
                              alt=""
                              bg={false}
                              className="mr2"
                              onImgLoad={[Function]}
                              src="/assets/images/seasonal-icon-color.svg"
                              style={Object {}}
                    />
                    <span>
                              Open Seasonally
                    </span>
          </div>
</div>,
      ],
      "className": "relative",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "z-0",
          "cluster": true,
          "clusterMaxZoom": 14,
          "clusterRadius": 50,
          "collections": Array [
            Object {
              "filter": Object {
                "ids": Array [],
              },
              "icon": "selected-location-icon",
              "name": "Selected",
            },
            Object {
              "filter": Object {
                "ids": Array [],
              },
              "icon": "seasonal-icon",
              "name": "SeasonalLocations",
            },
            Object {
              "filter": Object {
                "ids": undefined,
              },
              "name": "HiddenByFilter",
              "visible": false,
            },
          ],
          "defaultIcon": "year-round-icon",
          "featureCollection": Object {},
          "featureIdZoomTo": undefined,
          "hoverFade": false,
          "iconSize": 1.4,
          "mapPadding": 150,
          "maxZoom": 18,
          "onClickFeature": [Function],
          "onLoad": [Function],
          "styleUrl": "mapbox://styles/joshiefishbein/cjjyuj8fq0hrj2ro2j8066e4q",
          "textColor": "#ffffff",
          "textSize": 16,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              className="mr2 flex flex-wrap items-center drop-shadow"
              color="madison-blue"
              disabled={false}
              fullWidth={false}
              hover="shadow"
              label="All"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              shadow={false}
              to=""
              type="button"
              variant="primary-small"
/>,
            Array [],
          ],
          "className": "absolute t0 l0 flex p3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "className": "mr2 flex flex-wrap items-center drop-shadow",
              "color": "madison-blue",
              "disabled": false,
              "fullWidth": false,
              "hover": "shadow",
              "label": "All",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseLeave": [Function],
              "shadow": false,
              "to": "",
              "type": "button",
              "variant": "primary-small",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold"
>
              <Image
                            alt=""
                            bg={false}
                            className="mr2"
                            onImgLoad={[Function]}
                            src="/assets/images/year-round-icon-color.svg"
                            style={Object {}}
              />
              <span>
                            Open Year Round
              </span>
</div>,
            <div
              className="mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold"
>
              <Image
                            alt=""
                            bg={false}
                            className="mr2"
                            onImgLoad={[Function]}
                            src="/assets/images/seasonal-icon-color.svg"
                            style={Object {}}
              />
              <span>
                            Open Seasonally
              </span>
</div>,
          ],
          "className": "absolute b0 l0 flex p3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "false",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Image
                  alt=""
                  bg={false}
                  className="mr2"
                  onImgLoad={[Function]}
                  src="/assets/images/year-round-icon-color.svg"
                  style={Object {}}
/>,
                <span>
                  Open Year Round
</span>,
              ],
              "className": "mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "",
                  "bg": false,
                  "children": null,
                  "className": "mr2",
                  "onImgLoad": [Function],
                  "src": "/assets/images/year-round-icon-color.svg",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Open Year Round",
                },
                "ref": null,
                "rendered": "Open Year Round",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "true",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Image
                  alt=""
                  bg={false}
                  className="mr2"
                  onImgLoad={[Function]}
                  src="/assets/images/seasonal-icon-color.svg"
                  style={Object {}}
/>,
                <span>
                  Open Seasonally
</span>,
              ],
              "className": "mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "",
                  "bg": false,
                  "children": null,
                  "className": "mr2",
                  "onImgLoad": [Function],
                  "src": "/assets/images/seasonal-icon-color.svg",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Open Seasonally",
                },
                "ref": null,
                "rendered": "Open Seasonally",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MapboxMap
            className="z-0"
            cluster={true}
            clusterMaxZoom={14}
            clusterRadius={50}
            collections={
                        Array [
                                    Object {
                                      "filter": Object {
                                        "ids": Array [],
                                      },
                                      "icon": "selected-location-icon",
                                      "name": "Selected",
                                    },
                                    Object {
                                      "filter": Object {
                                        "ids": Array [],
                                      },
                                      "icon": "seasonal-icon",
                                      "name": "SeasonalLocations",
                                    },
                                    Object {
                                      "filter": Object {
                                        "ids": undefined,
                                      },
                                      "name": "HiddenByFilter",
                                      "visible": false,
                                    },
                                  ]
            }
            defaultIcon="year-round-icon"
            featureCollection={Object {}}
            featureIdZoomTo={undefined}
            hoverFade={false}
            iconSize={1.4}
            mapPadding={150}
            maxZoom={18}
            onClickFeature={[Function]}
            onLoad={[Function]}
            styleUrl="mapbox://styles/joshiefishbein/cjjyuj8fq0hrj2ro2j8066e4q"
            textColor="#ffffff"
            textSize={16}
/>,
          <div
            className="absolute t0 l0 flex p3"
>
            <Button
                        className="mr2 flex flex-wrap items-center drop-shadow"
                        color="madison-blue"
                        disabled={false}
                        fullWidth={false}
                        hover="shadow"
                        label="All"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        shadow={false}
                        to=""
                        type="button"
                        variant="primary-small"
            />
</div>,
          <div
            className="absolute b0 l0 flex p3"
>
            <div
                        className="mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold"
            >
                        <Image
                                    alt=""
                                    bg={false}
                                    className="mr2"
                                    onImgLoad={[Function]}
                                    src="/assets/images/year-round-icon-color.svg"
                                    style={Object {}}
                        />
                        <span>
                                    Open Year Round
                        </span>
            </div>
            <div
                        className="mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold"
            >
                        <Image
                                    alt=""
                                    bg={false}
                                    className="mr2"
                                    onImgLoad={[Function]}
                                    src="/assets/images/seasonal-icon-color.svg"
                                    style={Object {}}
                        />
                        <span>
                                    Open Seasonally
                        </span>
            </div>
</div>,
        ],
        "className": "relative",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "z-0",
            "cluster": true,
            "clusterMaxZoom": 14,
            "clusterRadius": 50,
            "collections": Array [
              Object {
                "filter": Object {
                  "ids": Array [],
                },
                "icon": "selected-location-icon",
                "name": "Selected",
              },
              Object {
                "filter": Object {
                  "ids": Array [],
                },
                "icon": "seasonal-icon",
                "name": "SeasonalLocations",
              },
              Object {
                "filter": Object {
                  "ids": undefined,
                },
                "name": "HiddenByFilter",
                "visible": false,
              },
            ],
            "defaultIcon": "year-round-icon",
            "featureCollection": Object {},
            "featureIdZoomTo": undefined,
            "hoverFade": false,
            "iconSize": 1.4,
            "mapPadding": 150,
            "maxZoom": 18,
            "onClickFeature": [Function],
            "onLoad": [Function],
            "styleUrl": "mapbox://styles/joshiefishbein/cjjyuj8fq0hrj2ro2j8066e4q",
            "textColor": "#ffffff",
            "textSize": 16,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                className="mr2 flex flex-wrap items-center drop-shadow"
                color="madison-blue"
                disabled={false}
                fullWidth={false}
                hover="shadow"
                label="All"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                shadow={false}
                to=""
                type="button"
                variant="primary-small"
/>,
              Array [],
            ],
            "className": "absolute t0 l0 flex p3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "className": "mr2 flex flex-wrap items-center drop-shadow",
                "color": "madison-blue",
                "disabled": false,
                "fullWidth": false,
                "hover": "shadow",
                "label": "All",
                "onClick": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "shadow": false,
                "to": "",
                "type": "button",
                "variant": "primary-small",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold"
>
                <Image
                                alt=""
                                bg={false}
                                className="mr2"
                                onImgLoad={[Function]}
                                src="/assets/images/year-round-icon-color.svg"
                                style={Object {}}
                />
                <span>
                                Open Year Round
                </span>
</div>,
              <div
                className="mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold"
>
                <Image
                                alt=""
                                bg={false}
                                className="mr2"
                                onImgLoad={[Function]}
                                src="/assets/images/seasonal-icon-color.svg"
                                style={Object {}}
                />
                <span>
                                Open Seasonally
                </span>
</div>,
            ],
            "className": "absolute b0 l0 flex p3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "false",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Image
                    alt=""
                    bg={false}
                    className="mr2"
                    onImgLoad={[Function]}
                    src="/assets/images/year-round-icon-color.svg"
                    style={Object {}}
/>,
                  <span>
                    Open Year Round
</span>,
                ],
                "className": "mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "alt": "",
                    "bg": false,
                    "children": null,
                    "className": "mr2",
                    "onImgLoad": [Function],
                    "src": "/assets/images/year-round-icon-color.svg",
                    "style": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Open Year Round",
                  },
                  "ref": null,
                  "rendered": "Open Year Round",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "true",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Image
                    alt=""
                    bg={false}
                    className="mr2"
                    onImgLoad={[Function]}
                    src="/assets/images/seasonal-icon-color.svg"
                    style={Object {}}
/>,
                  <span>
                    Open Seasonally
</span>,
                ],
                "className": "mr3 flex items-center justify-center pl1 pr2 text-dark-gray drop-shadow bg-white bold",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "alt": "",
                    "bg": false,
                    "children": null,
                    "className": "mr2",
                    "onImgLoad": [Function],
                    "src": "/assets/images/seasonal-icon-color.svg",
                    "style": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Open Seasonally",
                  },
                  "ref": null,
                  "rendered": "Open Seasonally",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
